cobra/util/solver.py

```diff
@@ -0,0 +1,170 @@
+                return solver_name
+    raise SolverNotFound("no mip-capable solver found")
+
+
+def add_to_solver(model, what=[]):
+    """Adds variables and constraints to a Model's solver object.
+
+    Useful for variables and constraints that can not be expressed with
+    reactions and lower/upper bounds. Will integrate with the Model's context
+    manager in order to revert changes upon leaving the context.
+
+    Parameters
+    ----------
+    model: a cobra model
+    The model to which to add the variables and constraints.
+    what: list or tuple of optlang variables or constraints.
+    The variables or constraints to add to the model. Must be of class
+    `model.solver.interface.Variable` or `model.solver.interface.Constraint`.
+    """
+    context = get_context(model)
+
+    if len(what) > 0:
+        model.solver.add(what)
+        if context:
+            context(partial(model.solver.remove, what))
+
+
+def remove_from_solver(model, what=[]):
+    """Removes variables and constraints from a Model's solver object.
+
+    Useful to temporarily remove variables and constraints from a Models's
+    solver object.
+
+    Parameters
+    ----------
+    model: a cobra model
+    The model from which to remove the variables and constraints.
+    what: list or tuple of optlang variables or constraints.
+    The variables or constraints to remove from the model. Must be of class
+    `model.solver.interface.Variable` or `model.solver.interface.Constraint`.
+    """
+    context = get_context(model)
+
+    if len(what) > 0:
+        model.solver.remove(what)
+        if context:
+            context(partial(model.solver.add, what))
+
+
+def add_absolute_expression(model, expression, name="abs_var", ub=None):
+    """Adds the absolute value of an expression to the model and defines
+    a variable for the absolute value that can be used in other objectives or
+    constraints.
+
+    Parameters
+    ----------
+    model: a cobra model
+    The model to which to add the absolute expression.
+    expression: A sympy expression
+    Must be a valid expression within the Model's solver object. The absolute
+    value is applied automatically on the expression.
+    name: string
+    The name of the newly created variable.
+    ub: positive float
+    The upper bound for the variable.
+    """
+
+    context = get_context(model)
+    variable = model.solver.interface.Variable(name, lb=0, ub=ub)
+
+    # The following constraints enforce variable > expression and
+    # variable > -expression
+    constraints = [
+        # positive value constraint
+        model.solver.interface.Constraint(expression - variable, ub=0,
+                                          name="abs_pos_" + name),
+        # negative value constraint
+        model.solver.interface.Constraint(expression + variable, lb=0,
+                                          name="abs_neg_" + name)
+        ]
+    model.solver.add(constraints + [variable])
```

Comment:
couldn't you just call `add_to_solver` above, and not duplicate the context code?